server/cluster/cluster.go

```diff
@@ -1879,19 +1879,21 @@ func (c *RaftCluster) onStoreVersionChangeLocked() {
 	failpoint.Inject("versionChangeConcurrency", func() {
 		time.Sleep(500 * time.Millisecond)
 	})
+	if minVersion == nil || clusterVersion.Equal(*minVersion) {
+		return
+	}
 
-	if minVersion != nil && clusterVersion.LessThan(*minVersion) {
-		if !c.opt.CASClusterVersion(clusterVersion, minVersion) {
-			log.Error("cluster version changed by API at the same time")
-		}
-		err := c.opt.Persist(c.storage)
-		if err != nil {
-			log.Error("persist cluster version meet error", errs.ZapError(err))
-		}
-		log.Info("cluster version changed",
-			zap.Stringer("old-cluster-version", clusterVersion),
-			zap.Stringer("new-cluster-version", minVersion))
+	if !c.opt.CASClusterVersion(clusterVersion, minVersion) {
```

Comment:
How about add some retry ?

Reply:
I'd like to keep the origin logic, which means if the cluster version changed by API at the same time, we should keep the version of API